lib/Unicode/Char/Case.hs

```diff
@@ -8,31 +8,42 @@
 -- Case and case mapping related functions.
 --
 module Unicode.Char.Case
-    ( isLower
+    ( -- * Predicates
+      isLowerCase
+    , isLower
+    , isUpperCase
     , isUpper
     )
 where
 
 import qualified Unicode.Internal.Char.DerivedCoreProperties as P
 
--- XXX Are these functions correct, as in do they match the behavior of that in base:Data.Char?
--- XXX Do we move these functions to Unicode.Char.Case.Compact?
-
 -- | Returns 'True' for lower-case letters.
 --
--- prop> isLower c == Data.Char.isLower c
+-- @since 0.3.0
+{-# INLINE isLowerCase #-}
+isLowerCase :: Char -> Bool
+isLowerCase = P.isLowercase
+
+-- | Returns 'True' for lower-case letters.
 --
 -- @since 0.1.0
 {-# INLINE isLower #-}
+{-# DEPRECATED isLower "Use isLowerCase instead. Note that the behavior of this function does not match base:Data.Char.isLower. See Unicode.Char.Case.Compat for behavior compatible with base:Data.Char." #-}
 isLower :: Char -> Bool
 isLower = P.isLowercase
 
```

Comment:
Deprecation may cause different behavior if someone imports `isLower` from `Unicode.Char.Case` vs from `Unicode.Char`, ideally both should give the same behavior.

Reply:
Maybe we could make a release without exposing the new `base`-compatible functions in `Unicode.Char`, then:
- bump the versión to 0.4.0,
- remove the deprecated functions and
- expose the`base`-compatible functions.

Reply:
Fixed